Fix to issue #33, 'fixed' pull request #35

Grimitsu opened the pull request:
Fixed issue #33 throwing BadQueryException. Some videos have keys xtag or axtags, which has value containing '=' on first attempt to get video. E.g. `axtags=tx=9417362,` though next attemps to get video contain no value value for this key i.e `axtags=`.

Also fixed KeyNotFoundException, two guys already tried to fix it #35 but that's not it.when video page provide no info about adaptive formats, i.e. there is no `adaptive_fmts` key instead there is a link to a MPD manifest with links to this videos and info about them.

[Change comparison](https://github.com/jamesqo/libvideo/compare/master...Grimitsu:master)

Grimitsu committed:
Case when no "adaptive_fmts" key provided

Grimitsu committed:
Method to parse MPD manifest

Grimitsu committed:
Add variant when adaptiveMap is empty

There is cases when there is no info about adaptive formats on the page, still it contains "dashmpd" key with a link to the XML file contatining metadata and links to the files.

Grimitsu committed:
Add case when Uris received from MPD

Grimitsu committed:
Remove equals check in query constructor

The only case when query constructor fails to get key=value pair is first attempt to get video. 
Query contains something like "axtags=tx=9417362" or "xtags=tx='number'".
Not sure for axtags, but xtags key present only in some non-adaptive videos e.g. format 248.
Next attempts to get video same video will return "xtags="

Grimitsu committed:
Update Query.cs

jamesqo commented on src/libvideo/Helpers/Html.cs:
```diff
@@ -14,5 +14,19 @@ internal static class Html
+                    .Select(v => v.Substring(0, v.IndexOf(closing)));
+                return Uris;
+            }
+            throw new NotSupportedException();
```
Question: Have you actually tried building/testing on your machine? It appears this is missing a closing brace. (To build you can open the `libvideo.sln` solution in VS + run NuGet restore and build.)

Grimitsu commented:
Please forgive me, James. I've messed things up trying to commit without unwanted code and comments and yes, i tried it, it works: libvideo.compat works, libvideo too. I'll fix my mistakes and make another commit.

Grimitsu committed:
Fix syntax mistakes, new debug program

Grimitsu commented:
@jamesqo fixed syntax mistakes. Modified debug program to test libvideo and libvideo.compat. Removed Newtonsoft.Json and YoutubeExtractor referencies from debug program and added libvideo.compat reference.(YoutubeExtractor namespace)

jamesqo commented on src/libvideo.debug/libvideo.debug.csproj:
```diff
@@ -61,6 +53,10 @@
     <None Include="packages.config" />
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\libvideo.compat\libvideo.compat.csproj">
```
Sorry, could you please revert the changes to this file? I referenced `YoutubeExtractor` instead of the compatibility API on purpose so I could flesh out any bugs in this lib vs. YoutubeExtractor.

Grimitsu committed:
Revert "Fix syntax mistakes, new debug program"

This reverts commit ae054cb7934b07409af4e299dc0b7e1732b08360.

Grimitsu committed:
Fix syntax mistakes, new debug program

Reverted changes to libvideo.debug.csproj

Grimitsu commented:
@jamesqo reverted changes to csproj. Please check [13422cf](https://github.com/jamesqo/libvideo/pull/39/commits/13422cf1c434e7d35dabc65d4def01681264554b)

jamesqo commented:
Merged in a new branch: https://github.com/jamesqo/libvideo/tree/merge-39

I'll publish the changes sometime tomorrow. Thanks!
